File: App/Static/scripts/ts/classes/entities/Board.ts
import {Column} from "./Column";

export class Board
{
	private readonly _id: number;
	private _name: string;
	
	private readonly _columns: Column[];
	
	constructor(id: number, name: string, columns: Column[])
	{
		this._id = id;
		this._name = name;
		this._columns = columns;
	}
	
	get columns(): Column[]
	{
		return this._columns;
	}
	
	get name(): string
	{
		return this._name;
	}
	
	set name(value: string)
	{
		this._name = value;
	}
}

File: DataAccessLayer/Entities/Board.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using DataAccessLayer.Entities.Constraints.Abstract;

namespace DataAccessLayer.Entities
{
	public class Board
	{
		public Board()
		{
			Labels      = new List<Label>();
			Columns     = new List<Column>();
			Constraints = new List<BoardConstraintEntityBase>();
		}

		public int Id { get; set; }

		[Required]
		[MaxLength(255)]
		public string Name { get; set; }

		public virtual Team Team   { get; set; }
		public         int  TeamId { get; set; }

		public virtual History History   { get; set; }
		public         int     HistoryId { get; set; }

		public virtual Project Project   { get; set; }
		public         int     ProjectId { get; set; }

		public virtual User Creator   { get; set; }
		public         int  CreatorId { get; set; }

		public virtual ICollection<Label>                     Labels      { get; set; }
		public virtual ICollection<Column>                    Columns     { get; set; }
		public virtual ICollection<BoardConstraintEntityBase> Constraints { get; set; }
	}
}